src/shared/generator.h

```diff
@@ -49,3 +51,17 @@ int generator_hook_up_growfs(
         const char *target);
 
 void log_setup_generator(void);
+
+/* Similar to DEFINE_MAIN_FUNCTION, but initializes logging and assigns positional arguments. */
+#define DEFINE_MAIN_GENERATOR_FUNCTION(impl)                            \
+        _DEFINE_MAIN_FUNCTION(                                          \
+                impl(argc > 1 ? argv[1] : "/tmp",                       \
+                     argc > 1 ? argv[2] : "/tmp",                       \
+                     argc > 1 ? argv[3] : "/tmp"),                      \
```

Comment:
the `argc >` checks are not right

Reply:
Can you be more specific? It seems to be the checks are equivalent to what was there before.

But I see one thing that should be changed for clarity: the order or arguments should be changed. I'll repush with such a change.

Reply:
Updated to change the order in macro.

Reply:
oh, i am an idiot, ignor me